nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/TestRepository.java

```diff
@@ -444,6 +449,68 @@ public void itCanHandleComplexPropertiesWithJson() throws IOException {
         assertEquals("file", document.getTitle());
     }
 
+    /**
+     * Dates can be handled properly if and only if the date is given as {@link java.lang.String}.
+     */
+    @Test
+    public void itCanHandleDateAsString() {
+        GregorianCalendar calendar = new GregorianCalendar(2017, Calendar.MAY, 4, 3, 2, 1);
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSXXX");
+        calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
+        formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+        String calendarStr = formatter.format(calendar.getTime());
+        assertEquals("2017-05-04T03:02:01.00Z", calendarStr);
+
+        Document birthday = new Document("Tom's Birthday", "Birthday");
```

Comment:
Why don't you use an already existing document type such as `File`? There is no added value to use a new type here.